packages/cardano-services/src/ChainHistory/DbSyncChainHistory/mappers.ts

```diff
@@ -215,13 +215,25 @@ export const mapTxAlonzo = (
     slot: Cardano.Slot(Number(txModel.block_slot_no))
   },
   body: {
+    ...(inputSource === Cardano.InputSource.collaterals
+      ? {
+          collateralReturn: outputs.length > 0 ? outputs[0] : undefined,
+          collaterals: inputs,
+          fee: BigInt(0),
+          inputs: [],
+          outputs: [],
+          totalCollateral: BigInt(txModel.fee)
```

Comment:
Maybe this is why I'm not still confident with these fields, but I would have said:

```suggestion
          collaterals: inputs,
          fee: BigInt(txModel.fee),
          inputs: [],
          outputs: [],
          totalCollateral: /* pseudo code: sum(inputs.coins) - outputs[0]?.coins */
```

Could you please explain me?
Thank you

Reply:
When the input source is collateral, db sync stores the collateral consumed as a fee, so we just need to map `fee` to `totalCollateral`

Reply:
AFAIK the target of this task is to make SDK compliant to Ledger rather than to db-sync.
IDK how a tx which fails phase 2 validation is represented at the lowest level:

- if it has distinct fee and collateral, IMHO we should make the distinction as well
- if everything is collected under the fee field, I'm fine with the change as is

Reply:
> AFAIK the target of this task is to make SDK compliant to Ledger rather than to db-sync.

That's correct

> if it has distinct fee and collateral, IMHO we should make the distinction as well
> if everything is collected under the fee field, I'm fine with the change as is

Some information is lost in db-sync, since the `fee` field is used for collateral amount. We are setting it to 0n as that is best we can do in this circumstance.